Microsoft.Toolkit/Deferred/DeferredEventArgs.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Toolkit.Deferred
+{
+    /// <summary>
+    /// <see cref="EventArgs"/> which can retrieve a <see cref="EventDeferral"/> in order to process data asynchronously before an <see cref="EventHandler"/> completes and returns to the calling control.
+    /// </summary>
+    public class DeferredEventArgs : EventArgs, IDisposable
```

Comment:
I tried to maintain history here, but git is so annoying about moving files and updating them... I even did them as two separate commits.